### PR TITLE
llvm target triple for Deno.build

### PR DESCRIPTION
**Pull request by ry**

### CODE REVIEW

File: cli/js/build.ts [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
-export const build: BuildInfo = {
-  arch: "" as Arch,
-  os: "" as OperatingSystem,
-};
-
-export function setBuildInfo(os: OperatingSystem, arch: Arch): void {
-  build.os = os;
+export function setBuildInfo(target: string): void {
+  const [arch, vendor, os, env] = target.split("-", 4);
+  //@ts-ignore
```
Comment (kitsonk) [inline cli/js/build.ts:14]: I don't think you would be getting errors here.

File: cli/js/build.ts [hunk 2]
```diff
@@ -1,24 +1,25 @@
 // Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
 
-export type OperatingSystem = "mac" | "win" | "linux";
+//@ts-ignore
```
Comment (kitsonk) [inline cli/js/build.ts:3]: I know you hate TS getting in the way, but why just not do:

```ts
export const build = {
  target: "unknown",
  arch: "unknown",
  os: "unknown",
  vendor: "unknown"
  env: undefined as (string | undefined)
};
```

Heck of a lot better than just ignoring errors.

Comment (kitsonk) [reply cli/js/build.ts:3]: Here is a [working example](https://www.typescriptlang.org/play/#code/KYDwDg9gTgLgBAYwgOwM7wEYFcCWAbAEzgF44BvAKDjhgEMoBzYGALjgCItkBrZCAd2TsANFTj0EACzacefQSLERUMrrwFDR1AG7BkBaKrkbF1PdrZcCwAGY5kwIrVRwAFOij2GcAD5wrtvaOAJQUAL4A3BQUoJCwcDZcCDA4KHCozABCuIQAksg2EK50jMxsHl7BbNoQOESU1NRIaPAA2hKSwnC6+tBdyl3mALokNPRMMAB0qGB4ODCu7AC0InAALMFRjXDY+ASTJROjh8xbjbuEkx2jHWfUF-s9BlCjT9B3Ozn7yqPKHw+TcyjcxRMIUIA) in strict mode with no ignoring of errors and no real "trickery" except for the env.

Comment (ry) [reply cli/js/build.ts]: Thanks - fixed.

File: cli/js/build.ts [hunk 3]
```diff
@@ -1,24 +1,19 @@
...
-  arch: Arch;
-
-  os: OperatingSystem;
-}
-
-export const build: BuildInfo = {
-  arch: "" as Arch,
-  os: "" as OperatingSystem,
+export const build = {
+  target: "unknown",
```
Comment (ry) [inline cli/js/build.ts:4]: Rename to Deno.build.triple ?

Comment (bartlomieju) [reply cli/js/build.ts:4]: I think `target` is fine

**Pull request merged**